src/test/mod.rs

```diff
@@ -220,3 +186,410 @@ fn structs() {
         *engine.call(&main_func, Vec::new()).unwrap()
     );
 }
+
+mod variadic_functions {
+    use freight_vm::error::FreightError;
+
+    use super::*;
+
+    #[test]
+    fn fixed_args() {
+        let script = r#"
+            $fName = (a, b, c){}
+            fName(1, 2, 3)
+            fName(1, 2, 3, 4)
+            null
+        "#;
+
+        let (mut engine, main_func) = crate::interpreter::create_engine_main(script).unwrap();
```

Comment:
Why are you doing it this way instead of using `run()`?

Reply:
now uses `run_error`